crates/polars-arrow/src/trusted_len/boolean.rs

```diff
@@ -30,3 +31,54 @@ impl FromTrustedLenIterator<bool> for BooleanArray {
         }
     }
 }
+
+impl FromIteratorReversed<bool> for BooleanArray {
+    fn from_trusted_len_iter_rev<I: TrustedLen<Item = bool>>(iter: I) -> Self {
+        let size = iter.size_hint().1.unwrap();
+
+        let mut vals: Vec<bool> = Vec::with_capacity(size);
+        unsafe {
```

Comment:
Can we directly set the bits on a `Vec<u8>` here? That would save us going from `Vec<bool>` which a a byte per boolean and then copying that into the bitmaks of a bit per boolean.

I believe we already do this correctly in the code below.